lib/model.js

```diff
@@ -1857,21 +1857,13 @@ class Model {
       countOptions.attributes = undefined;
     }
 
-    const countQuery = this.count(countOptions);
-    const findQuery = this.findAll(options);
-
-    return countQuery.then(count => {
```

Comment:
This used to be a performance improvement if count is zero then we can resolve empty array in rows and count to zero. I would like to keep that behavior

Reply:
First of all, it is obvious that the current solution does not work, regardless of performance. It should be fixed.
Besides, I believe that the findQuery will be executed very quickly if count is zero and there is no point in talking about reducing productivity.
Perhaps if this worries you, you have the convenient means to test performance?